test/e2e/clusterclass_changes.go

```diff
@@ -177,7 +177,14 @@ func ClusterClassChangesSpec(ctx context.Context, inputGetter func() ClusterClas
 			Cluster:                   clusterResources.Cluster,
 			WaitForMachineDeployments: input.E2EConfig.GetIntervals(specName, "wait-worker-nodes"),
 		})
+		By("PASSED!")
 
+		By("Deleting a MachineDeploymentTopology in the Cluster Topology and wait for associated MachineDeployment to be deleted")
```

Comment:
Tested locally - passing on main and failing on main before the fix, so it looks like it should correctly catch a similar bug in future.